source/PlayerInfo.cpp

```diff
@@ -2354,7 +2364,34 @@ void PlayerInfo::StepMissions(UI *ui)
 		else if(mission.CanComplete(*this))
 			RemoveMission(Mission::COMPLETE, mission, ui);
 		else if(mission.Destination() == GetPlanet() && !freshlyLoaded)
+		{
 			mission.Do(Mission::VISIT, *this, ui);
+			if(mission.IsUnique())
+				continue;
+			
+			// On visit dialogs are handled separately as to avoid a player
+			// getting spammed by on visit dialogs if they are stacking jobs
+			// from the same destination.
+			auto action = mission.GetAction(Mission::VISIT);
+			if(!action.DialogText().empty() && visitText.empty())
+				visitText = Format::Replace(action.DialogText(), substitutions);
+			++missionVisits;
+			if(mission.Deadline())
+				++deadlineMissions;
+		}
+	}
+	if(!visitText.empty())
+	{
+		visitText += "\n\tYou have " + Format::Number(missionVisits) + " unfinished " 
```

Comment:
If we have only 1 repeating mission with a deadline and a visit dialog, (and no others at this location) we see something like:
```
<that mission's visit text>

You have 1 unfinished mission at this location (1 of which has an approaching deadline).
```

Should we even show the counts for a single mission? I feel like it should be an "other" snippet, e.g. `(You have N other unfinished missions at this location.)` and we shouldn't include any information about the job whose text we showed, even if it has a deadline. Probably that information, if relevant for the player, should have already been included in the mission's visit text.

Reply:
Yeah, I caught the fact that this shows up regardless, but hadn't implemented anything to avoid showing this extra bit if you only have one on visit.